### ISSUE TITLE
Beta diversity app: refactor to better fit compute plugin

### ISSUE THREAD

**Issue by asizemore** [VEuPathDB/microbiomeComputations#16]
Revisit beta div now that we have some alpha div and ranked abundance experience. Make sure it's everything we need.

**Comment by asizemore** [comment]
beta div needs to be organized so that it's easy to get the tabular data, and then the metadata as json. Possibly use two functions?

**Comment by asizemore** [comment]
In more detail, i think it'll be nicer to have the main beta div function return the data table with a bunch of attributes. Then we can have `writeDataResult` in the compute plugin write just the data table. Once that's done, a second function can take the results object (data table with lots of attributes) and pull out just the nice json-formatted metadata. We can write that metadata with `writeMetaResult` in the compute plugin.

Basically since the writing of results is broken into two steps in the compute plugin, it'd be easier to have it split up here as well.
